rust/signed_doc/src/metadata/uuid_type/mod.rs

```diff
@@ -0,0 +1,27 @@
+//! `UUID` types.
```

Comment:
I will get a new crate made for general types like this to live.  Keep it for now, but refactor once its merged.

However, we should make types for each of the metadata feilds that are UUID, so the types define what they are, such as;

```
pub struct DocTypeId(Uuidv4)
```
And we use that where we want a `DocTypeId` not just `Uuidv4` it will make a lot of the code clearer about whats in the type, especially in places where we have Tuples like `(UuidV7, UuidV7)` where its inferred what they are, but not actually specified by the types.